corehq/apps/userreports/expressions/specs.py

```diff
@@ -79,8 +79,18 @@ def configure(self, context):
             raise BadSpecError(u'Name {} not found in list of named expressions!'.format(self.name))
         self._context = context
 
+    def _context_cache_key(self):
+        return 'named_expression-{}'.format(self.name)
```

Comment:
It looks like this key would need more information in it, but maybe not because this uses the iteration cache? I'm not super familiar with how named expressions work, but what would happen if you used a named expression as some sort of sub-expression. Would the "item" be different?